### PR TITLE
Bug Fix and enhancments

### PR DESCRIPTION
**Pull request by KozmikNano**
My goal is to fix and close as many bugs, and enhancements! 

# Do Not Merge

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const antiafk = require("mineflayer-antiafk");
 const queueData = require("./queue.json");
 const util = require("./util");
 const save = "./saveid";
+const http = require("http")
```
Comment (rob9315) [inline main.js:23]: why

Comment (KozmikNano) [reply main.js:23]: It's needed to check if 2b2t.io is reachable

Comment (KozmikNano) [reply main.js:23]: It also want called before

File: main.js [hunk 2]
```diff
@@ -258,12 +259,15 @@ function join() {
...
-								logActivity("P: " + positioninqueue + " E: " + webserver.ETA + " - " + client.username);}
-						}} else {
+						if (config.get("userStatus")) {
+							{ //set the Discord Activity
+								if (displayEmail) {
+									logActivity("P: " + positioninqueue + " E: " + webserver.ETA + " - " + options.username);
+								} else {
+									logActivity("P: " + positioninqueue + " E: " + webserver.ETA + " - " + client.username);
+								}
+							}
```
Comment (rob9315) [inline main.js:269]:
```suggestion
						if (config.get("userStatus")) {
							//set the Discord Activity
							if (displayEmail) {
								logActivity("P: " + positioninqueue + " E: " + webserver.ETA + " - " + options.username);
							} else {
								logActivity("P: " + positioninqueue + " E: " + webserver.ETA + " - " + client.username);
							}
```

File: main.js (hunk 1 above)
Comment (rob9315) [reply main.js:23]:
```suggestion
```

**Review by rob9315 — CHANGES_REQUESTED** [review]
change these

File: main.js (hunk 2 above)
Comment (KozmikNano) [reply main.js:269]: Whats the difference?

Comment (KozmikNano) [reply main.js:268]: Oh wait i see it now
I wonder why i have been seeing a lot of duplicate stuff lately.

**Review by rob9315 — APPROVED** [review]

**Pull request merged**